libs/domain/cart/src/mixins/cart.mixin.ts

```diff
@@ -1,22 +1,53 @@
 import { Type } from '@spryker-oryx/di';
-import {
-  ComponentMixin,
-  ContentComponentProperties,
-} from '@spryker-oryx/experience';
+import { asyncState, valueType } from '@spryker-oryx/utilities';
 import { LitElement } from 'lit';
 import { property } from 'lit/decorators.js';
-import type { Cart, CartComponentAttributes } from '../models';
-
-export const CartComponentMixin = <T>(): Type<
-  LitElement & ContentComponentProperties<T>
-> => {
-  class CartComponent
-    extends ComponentMixin<T>()
-    implements CartComponentAttributes
-  {
-    @property({ type: String }) cartId?: string;
-    @property({ type: Object }) cart?: Cart;
-  }
+import { CartController } from '../controllers';
+import type {
+  Cart,
+  CartComponentAttributes,
+  CartEntry,
+  FormattedCartTotals,
+} from '../models';
+
+export declare class CartMixinInterface implements CartComponentAttributes {
+  cartId?: string;
+  cart?: Cart;
+
+  protected cartController: CartController;
+
+  protected isEmpty?: boolean;
+  protected isBusy?: boolean;
+  protected entries?: CartEntry[];
+  protected totals?: FormattedCartTotals;
+  protected totalQuantity?: number;
+}
+
+export const CartComponentMixin = <
+  T extends Type<LitElement & CartComponentAttributes>
+>(
+  superClass: T
+): Type<CartMixinInterface> & T => {
+  class CartMixinClass extends superClass {
+    @property({ reflect: true }) cartId?: string;
+    @property({ type: Object, reflect: true }) cart?: Cart;
+
+    protected cartController = new CartController(this);
 
-  return CartComponent as Type<LitElement & ContentComponentProperties<T>>;
+    @asyncState()
```

Comment:
Debatable, but I think that we should define protected observable first and then pass it to the asyncState. This allows to have a bit more flexibility.

```
protected isEmpty$ = this.cartController.isEmpty();

@asyncState()
protected isEmpty = valueType(this.isEmpty$);

...

```

Reply:
I don't see the issue, you might want to challenge @dunqan outside the scope of this refactor as it's part of cart service refactor done earllier (just following a pattern here). Also, we're movig to signals which will change things again slightly.